src/control/positionControl/pathPlanning/VoronoiPathPlanning.cpp

```diff
@@ -0,0 +1,153 @@
+        }
+    }
+    if (graphAdjacencyList.find(robotPosition) == graphAdjacencyList.end()) {
+        graphAdjacencyList.insert(
+                {robotPosition, std::list<GraphNode>(1, (GraphNode) {closestOriginPoint, minOriginDist})});
+    }
+    if (graphAdjacencyList.find(targetPosition) == graphAdjacencyList.end()) {
+        graphAdjacencyList[closestTargetPoint].push_back({targetPosition, minTargetDist});
+    }
+}
+
+const std::unordered_map<rtt::Vector2, std::list<GraphNode>, hashPoint> &
+VoronoiPathPlanning::getGraphAdjacencyList() const {
+    return graphAdjacencyList;
+}
+
+std::list<rtt::Vector2> VoronoiPathPlanning::generatePathDijkstra(const rtt::Vector2& initialPosition, const rtt::Vector2& targetPosition) {
+    std::unordered_map<rtt::Vector2, float, hashPoint> distanceVector;
+    std::unordered_map<rtt::Vector2, rtt::Vector2, hashPoint> parentVector;
+    distanceVector.insert({initialPosition,0});
+    std::list<rtt::Vector2> nodeQueue;
+    nodeQueue.push_front(initialPosition);
+    for(const rtt::Vector2& currentNode: nodeQueue){
+        if (currentNode == targetPosition){
+            break;
+        }
+        for (const GraphNode& adjacentNode : graphAdjacencyList[currentNode]){
+            if (distanceVector.find(adjacentNode.nextNodePosition) == distanceVector.end()) {
+                if (std::find(nodeQueue.begin(), nodeQueue.end(), adjacentNode.nextNodePosition) == nodeQueue.end()) {
+                    nodeQueue.push_back(adjacentNode.nextNodePosition);
+                }
+                distanceVector.insert({adjacentNode.nextNodePosition, distanceVector[currentNode] + adjacentNode.distance});
+                parentVector.insert({adjacentNode.nextNodePosition, currentNode});
+                continue;
+            }
+            if(distanceVector[adjacentNode.nextNodePosition] > distanceVector[currentNode] + adjacentNode.distance){
+                distanceVector[adjacentNode.nextNodePosition] = distanceVector[currentNode] + adjacentNode.distance;
+                parentVector[adjacentNode.nextNodePosition] = currentNode;
+            }
+        }
+    }
+
+    std::list<rtt::Vector2> pathPoints;
+    for (rtt::Vector2 backtrack = targetPosition; parentVector.find(backtrack) != parentVector.end(); backtrack = parentVector[backtrack]){
+        pathPoints.push_back(backtrack);
+    }
+    std::reverse(pathPoints.begin(), pathPoints.end());
+    return pathPoints;
+}
+
+std::list<rtt::Vector2> VoronoiPathPlanning::computePath(const rtt::Vector2 &robotPosition, const rtt::Vector2 &targetPosition) {
+    computeDiagram(robotPosition, targetPosition);
+    generateGraphFromDiagram();
+    completeGraphWithOriginDestination(robotPosition, targetPosition);
+    return generatePathDijkstra(robotPosition, targetPosition);
+}
+
+rtt::Vector2 VoronoiPathPlanning::convertFromJcvPoint(jcv_point point) {
+    return rtt::Vector2(std::round(1000.0*point.x)/1000, std::round(1000.0*point.y)/1000);
+}
```

Comment:
Also, why in the hell would I round my point precision down? This function needs a comment to why it even exist and how this  'converts' since you are losing precision putting things through here.

Reply:
I wanted to round it to 3 decimals for testing purposes. It's not like anything smaller than 1mm is relevant